build/charts/theia/templates/clickhouse/nfs-persistentvolume.yaml

```diff
@@ -1,16 +1,25 @@
 {{- if eq .Values.clickhouse.storage.createPersistentVolume.type "NFS" }}
+{{ $pvNum := 1 }}
+{{- if .Values.clickhouse.cluster.enable }}
+{{ $pvNum = int (mul .Values.clickhouse.cluster.shards .Values.clickhouse.cluster.replicas) }}
+{{- end }}
+{{- range $i, $_ := until $pvNum }}
+{{- if not (eq $i 0) }}
+---
+{{- end }}
 apiVersion: v1
 kind: PersistentVolume
 metadata:
-  name: clickhouse-pv
+  name: clickhouse-pv-{{$i}}
```

Comment:
If I'm not mistaken, we are creating every DB on the same PV. What's the reasoning for this?

As a user I'm using replication for performance and fault tolerance. If all the replicas are on the same PV I hardly get any fault tolerance (for data loss), and also performance might not be that much different from single replica as all the I/O is still occurring on the same device.

Did you consider an option with a distinct PV per replica?

Reply:
To be clear, we are creating every replica on different PV, but all the PVs are on the same node for Local case, and the same NFS server for the NFS case. As users can use `persistentVolumeClaimSpec` for customized PVs, I plan to provide a simple case for the Local and NFS solution.

I'm thinking about if I introduce multiple nodes or NFS servers, then there might be cases that users do not have enough nodes or NFS servers, and they may want to have different PV numbers on each node or NFS servers, which makes it much more complex.

But using a distinct Node or NFS server per replica is not that complex. Your consideration about the fault tolerance and performance makes sense. It might be a better default case for our Local/NFS PV solution. Let me switch to this.

Reply:
Now each replica will use distinct Node or NFS server.